libnetwork/types/network.go

```diff
@@ -213,6 +213,9 @@ type PerNetworkOptions struct {
 	// InterfaceName for this container. Required in the backend.
 	// Optional in the frontend. Will be filled with ethX (where X is a integer) when empty.
 	InterfaceName string `json:"interface_name"`
+	// Path to WireGuard configuration.
+	// Will only be used when the network driver is WireGuard
+	WireGuardConfigPath string `json:"wireguard_config_path"`
```

Comment:
This is IMO not sustainable, we should not add driver specific options here. This is exposed via podman API.

I think we should add a generic options field as map[string]string where we could in future add more driver specific fields Without having to update the struct every single time.

Reply:
Similar to what we're doing in containers.conf for runtimes?

https://github.com/containers/common/blob/main/pkg/config/config.go#L358-L359

Reply:
Updated this with `map[string]string`. Multiple values could still be separated by a comma and parsed before they are used